### ISSUE TITLE
Wrong transformation on mapped item

### ISSUE THREAD

**Issue by hlg** [opensourceBIM/BIMserver#1127]
This one is observed on the stair flights in the Schependomlaan data set [1], extracted sample attached below [2].

![stairs-bimserver_s](https://user-images.githubusercontent.com/356986/76528606-619d4400-6471-11ea-97b6-06a07000c447.png) ![stairs-fzk_s](https://user-images.githubusercontent.com/356986/76528808-acb75700-6471-11ea-87a1-da65bb75806f.png)

The left (violet) is the geometry in BIMServer, while the right (orange) is in FZK viewer.

The two flights have mapped representations with the same geometry, and varying placments in both the representation mapping and the product itself.

To quickly check whether it is an IfcOpenShell issue, I converted the sample file to obj with IFCconvert, which looks fine.

[1] [IFC Schependomlaan.ifc](https://github.com/openBIMstandards/DataSetSchependomlaan/blob/master/Design%20model%20IFC/IFC%20Schependomlaan.ifc)
[2] [Stairs-2-storey.ifc.txt](https://github.com/opensourceBIM/BIMserver/files/4324556/Stairs-2-storey.ifc.txt)

**Comment by rubendel** [comment]
I think I fixed this by uncommenting some commented lines. Not sure why those were commented in the first place. My knowledge on this subject (matrices etcetera) is basically non-existing, so a review would be appreciated.

**Comment by hlg** [comment]
Although this works for the case given, the code looks wrong. According to how I understand the algorithm given in the IFC spec, the if branches should be switched. Maybe that is why it was commented, because some other case did not work with the accidentally swapped condition? Further I don't really understand why the fix works, because it only changes something for `axis1 == null` - which is not the case for the misplaced flight.

It must have to do with mapped item optimization - switching this option off leads to proper display in the first place. Which is also surprising (probably a coincidence), because I found a more severe error: IfcCoordinateTransformation does not have an Axis3 attribute, this is only there for IfcCoordinateTransformation3D. Thus, `axis3` was always null and `a3` defaulted to `[0,0,1]`. After fixing that it looks fine even with mapped item optimization, at least with this stairflight sample.

During the review I also found that IfcCoordinateTransformation3D scale is ignored (probably not used often) and that there is an error in calculating the matrix from IfcAxis2Placement3D. The algorithm here assumes perpendicular axis and refAxis which is the case in most samples, but actually not required.
